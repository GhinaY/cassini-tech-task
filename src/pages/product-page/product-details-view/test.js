import { render } from '@testing-library/react';
import mockListingsData from '../../../utils/test-utils/mock-listings-data';
import ProductDetailsView from '.';
import { MemoryRouter } from 'react-router-dom';

function renderComponent(props) {
  return(
    render(
      <MemoryRouter>
        <ProductDetailsView {...props}/>
      </MemoryRouter>
    )
  );
};

describe('ProductDetailsView', () => {
  const mockListingDetails = mockListingsData[0];
  const mockSetIsEditing = jest.fn();

  test('renders correctly when listing data is available', () => {
    const { asFragment } = renderComponent({ listingDetails: mockListingDetails, setIsediting: mockSetIsEditing });

    expect(asFragment()).toMatchSnapshot();
  });

  test('renders error message when listing data is not available', () => {
    const { asFragment } = renderComponent({ setIsediting: mockSetIsEditing });

    expect(asFragment()).toMatchSnapshot();
  });
});

// Definitely needs a lot more tests, including functional ones, 
// but in the interests of time I'll skip them for now.
