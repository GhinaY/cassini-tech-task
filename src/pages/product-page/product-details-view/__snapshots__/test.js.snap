// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetailsView renders correctly when listing data is available 1`] = `
<DocumentFragment>
  <img
    alt="the product Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops"
    class="ProductImage"
    src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
  />
  <div
    class="ProductDetails"
  >
    <a
      class="BackButton"
      href="/"
    >
      &lt; Back to listings
    </a>
    <span
      class="ProductCategory"
    >
      men's clothing
    </span>
    <h1
      class="ProductTitle"
    >
      Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops
    </h1>
    <span>
      ID: 1
    </span>
    <span>
      Â£109.95
    </span>
    <p
      class="ProductDescription"
    >
      Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday
    </p>
    <button
      class="Button"
    >
      Edit
    </button>
  </div>
</DocumentFragment>
`;

exports[`ProductDetailsView renders error message when listing data is not available 1`] = `
<DocumentFragment>
  <div>
    No product information available
  </div>
</DocumentFragment>
`;
